devel/addon/certmanager/install.sh

```diff
@@ -38,7 +38,13 @@ check_tool helm
 # triggered on every call to this script.
 export APP_VERSION="$(date +"%s")"
 # Build a copy of the cert-manager release images using the :bazel image tag
-bazel run --stamp=true --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64 "//devel/addon/certmanager:bundle"
+
+ARCH="$(uname -m)"
+if [ "$ARCH" == "arm64" ] ; then
```

Comment:
Thought: the title of the PR is build for the host architecture, but what this does is check if uname -m is arm64 and build it for linux/arm64, otherwise build amd64.

If we want to build for the host architecture, should we try and load the [toolchain that matches](https://github.com/bazelbuild/rules_go/blob/728a9e1/go/toolchains.rst#go-toolchain suggests) the host architecture? Although I can't think of how to do this off the top of my head if a go sdk is not installed already (I would use ` go env GOARCH`.

Reply:
Pragmatically, it is an improvement so could be merged, but perhaps with a title change 😅

Reply:
We register the go toolchain in WOKSPACE https://github.com/jetstack/cert-manager/blob/master/WORKSPACE#L37-L40

But this `bazel run` ends up calling a rules that aren't Go rules (I think first it goes [here](https://github.com/jetstack/cert-manager/blob/master/build/BUILD.bazel#L48-L75) where the target architecture value might be used to as an argument)- so I'm not sure if it would use the GOARCH value in the same way? I wonder if whether that flag was skipped altogether it would just autodetect the architecture correctly.

Reply:
Good point, I've renamed the PR.

Reply:
I agree that automatically detecting arch and loading the toolchain would be cleaner, but I don't know enough about bazel and how we've configured it to make that work.

I did attempt this by removing the `--platforms` arg but it fails with the output below. I think we'd need to dig a little deeper to resolve this and I am not sure it's worth it if we are moving away from bazel. 

```
❯ bazel run --stamp=true "//devel/addon/certmanager:bundle"
Starting local Bazel server and connecting to it...
(10:48:18) INFO: Current date is 2022-01-04
(10:48:25) INFO: SHA256 (https://golang.org/dl/?mode=json&include=all) = 82da1d733c88951b99e91ef86ea88d469df31968672db14f163a2a4b7faee92e
(10:48:31) ERROR: /Users/joakim/git/cert-manager/build/BUILD.bazel:50:22: Configurable attribute "architecture" doesn't match this configuration (would a default condition help?).
Conditions checked:
 @io_bazel_rules_go//go/platform:linux_amd64
 @io_bazel_rules_go//go/platform:linux_arm64
 @io_bazel_rules_go//go/platform:linux_arm
 @io_bazel_rules_go//go/platform:linux_ppc64le
 @io_bazel_rules_go//go/platform:linux_s390x
(10:48:31) ERROR: Analysis of target '//devel/addon/certmanager:bundle' failed; build aborted: /Users/joakim/git/cert-manager/build/BUILD.bazel:50:22: Configurable attribute "architecture" doesn't match this configuration (would a default condition help?).
Conditions checked:
 @io_bazel_rules_go//go/platform:linux_amd64
 @io_bazel_rules_go//go/platform:linux_arm64
 @io_bazel_rules_go//go/platform:linux_arm
 @io_bazel_rules_go//go/platform:linux_ppc64le
 @io_bazel_rules_go//go/platform:linux_s390x
(10:48:31) INFO: Elapsed time: 15.962s
(10:48:31) INFO: 0 processes.
(10:48:31) FAILED: Build did NOT complete successfully (21 packages loaded, 123 targets configured)
(10:48:31) FAILED: Build did NOT complete successfully (21 packages loaded, 123 targets configured)

```

Reply:
So this is because we don't want `darwin_arm64` images built for the `linux_arm64` OS that kind runs in (i.e. docker for mac on M1).

We can certainly look at tweaking the `multi_arch_container_build` macro that @irbekrm linked to build _always_ for linux, but respect the host's architecture. However the tricky part is that we also have dependencies on other software in the 'devel' environment, and this software/images are not multiarch either. This change will mean *cert-manager* is built/tested with the same architecture as the host, but the VM itself still needs qemu etc to handle emulating amd64 in the arm64 environment anyway.

Reply:
> If we want to build for the host architecture, should we try and load the toolchain that matches the host architecture? Although I can't think of how to do this off the top of my head if a go sdk is not installed already (I would use go env GOARCH.)

There's an example of this in the kind repo (prior to kind switching to use 'gimme' to install go):

https://github.com/kubernetes-sigs/kind/blob/86a075eeefbd0a4550581534db8c8ce6ebc56660/hack/go_container.sh#L40-L72